modules/javafx.controls/src/main/java/javafx/scene/control/Control.java

```diff
@@ -222,8 +222,14 @@ private static Class<?> loadClass(final String className, final Object instance)
      * {@code Skin}. Every {@code Skin} maintains a back reference to the
      * {@code Control} via the {@link Skin#getSkinnable()} method.
      * <p>
+     * To ensure a one-to-one relationship between a {@code Control} and its {@code Skin},
+     * {@link Control#setSkin(Skin)} will check the return value of {@link Skin#getSkinnable()}
+     * against this Control, and throw an {@code IllegalArgumentException} if it is not the same.
+     * <p>
      * A skin may be null.
+     *
      * @return the skin property for this control
+     * @throws IllegalArgumentException if {@code skin != null && skin != getSkinnable()}
      */
```

Comment:
hmm ..  shouldn't the condition be `(skin != null && skin.getSkinnable() != Control.this)`?

Reply:
The check is actually inside of a StyleableObjectProperty, so Control.this is appropriate (see line 250), though I think the human-readable explanations in the same javadoc comment might be clear enough (I hope they are clear enough)